src/python/pants/backend/javascript/goals/test.py

```diff
@@ -0,0 +1,245 @@
+        output_directories.extend(test_script.coverage_output_directories)
+        entry_point = test_script.coverage_entry_point or entry_point
+
+    process = await Get(
+        Process,
+        NodeJsProjectEnvironmentProcess(
+            installation.project_env,
+            args=(
+                "run",
+                entry_point,
+                "--",
+                *sorted(map(relative_package_dir, field_set_source_files.files)),
+                *coverage_args,
+            ),
+            description=f"Running npm test for {field_set.address.spec}.",
+            input_digest=merged_digest,
+            level=LogLevel.INFO,
+            extra_env=FrozenDict(**test_extra_env.env, **target_env_vars),
+            timeout_seconds=field_set.timeout.calculate_from_global_options(test),
+            output_files=tuple(output_files),
+            output_directories=tuple(output_directories),
+        ),
+    )
+    if test.force:
+        process = dataclasses.replace(process, cache_scope=ProcessCacheScope.PER_SESSION)
+
+    result = await Get(FallibleProcessResult, Process, process)
+    coverage_data: JSCoverageData | None = None
+    if test.use_coverage:
+        coverage_snapshot = await Get(
+            Snapshot,
+            DigestSubset(
+                result.output_digest,
+                test_script.coverage_globs(installation.project_env.relative_workspace_directory()),
+            ),
+        )
+        coverage_data = JSCoverageData(
+            coverage_snapshot,
+            field_set.address,
+            output_files=test_script.coverage_output_files,
+            output_directories=test_script.coverage_output_directories,
+            working_directory=installation.project_env.relative_workspace_directory(),
+        )
+
+    return TestResult.from_fallible_process_result(
+        result, field_set.address, test.output, coverage_data=coverage_data
+    )
+
+
+@rule(desc="Collecting coverage reports.")
+async def collect_coverage_reports(
+    coverage_reports: JSCoverageDataCollection,
+    dist_dir: DistDir,
+    nodejs_test: NodeJSTest,
+) -> CoverageReports:
+    gets_per_data = [
+        (
+            file,
+            report,
+            Get(
+                Snapshot,
+                DigestSubset(
+                    report.snapshot.digest,
+                    NodeTestScript.coverage_globs_for(
+                        report.working_directory,
+                        (file,),
+                        report.output_directories,
+                        GlobMatchErrorBehavior.error,
+                        GlobExpansionConjunction.all_match,
+                        description_of_origin="the JS coverage report collection rule",
+                    ),
+                ),
+            ),
+        )
+        for report in coverage_reports
+        for file in report.output_files
+    ]
+    snapshots = await MultiGet(get for file, report, get in gets_per_data)
```

Comment:
This parses strangely when the unused fields are named. Maybe:
```suggestion
    snapshots = await MultiGet(get for _, _, get in gets_per_data)
```